include/odp/api/spec/packet.h

```diff
@@ -401,30 +401,39 @@ uint32_t odp_packet_buf_len(odp_packet_t pkt);
 /**
  * Packet data pointer
  *
- * Returns the current packet data pointer. When a packet is received
- * from packet input, this points to the first byte of the received
- * packet. Packet level offsets are calculated relative to this position.
+ * Returns pointer to the first byte of packet data. When packet is segmented,
+ * only a portion of packet data follows the pointer. When unsure, use e.g.
+ * odp_packet_seg_len() to check the data length following the pointer. Packet
+ * level offsets are calculated relative to this position.
  *
- * User can adjust the data pointer with head_push/head_pull (does not modify
- * segmentation) and add_data/rem_data calls (may modify segmentation).
+ * When a packet is received from packet input, this points to the first byte
+ * of the received packet. Pool configuration parameters may be used to ensure
+ * that the first packet segment contains all/most of the data relevant to the
+ * application.
```

Comment:
I was wondering if the above statement on pool configuration is accurate and checked that pool configuration documentation says this:

                /** Minimum number of packet data bytes that are stored in the
                 *  first segment of a packet. The maximum value is defined by
                 *  pool capability pkt.max_seg_len. Use 0 for default.
                 */
                uint32_t seg_len;

I think it is unclear what _exactly_ is meant by the above. For instance, right after allocating a new packet, the first segment contains zero data bytes regardless of the seg_len setting. I suppose the intent is that the first segment contains the minimum of seg_len and odp_packet_len() data bytes. But when does that condition hold? After receiving a packet from pktio (yes)? After receiving a packet from a queue (dunno)? After extending or truncating or otherwise manipulating the packet buffer (probably not)?

I would suggest clarifying the documentation of the seg_len pool parameter. This is related to this patch indirectly through the reference to the pool configuration.

Reply:
"Minimum number of packet data bytes that are stored in the ... " would be clearer with "... that can be stored ... ".

Pool config ensures that packet has large enough first segment for storing seg_len bytes. Packet input documentation could state that packet input always fills the first segment fully.

When application manipulates packet, various segment sizes may result.